homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -179,17 +235,75 @@ async def async_set_cover_position(self, **kwargs):
 
     async def _async_move(self, target_hass_position):
         """Move the shade to a position."""
-        current_hass_position = hd_position_to_hass(self._current_cover_position)
+        time_delta = datetime.now(tz=None) - timedelta(minutes=5)
+        # we will limit force update to 5 minutes to prevent spamming hub
+        force_refresh = False
+        if self._last_forced_refresh is not None:
+            force_refresh = self._last_forced_refresh < time_delta
+
+        if self._motor in ["Top", "Bottom"]:
```

Comment:
This code is too complex to follow.  Please move top and bottom shades into their own class so we can split the code